allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -338,8 +338,18 @@ class S3CloudCache(CloudCacheBase):
     ----------
     cache_dir: str or pathlib.Path
         Path to the directory where data will be stored on the local system
+
+    bucket_name: str
+        for example, if bucket URI is 's3://mybucket' this value should be
+        'mybucket'
+
     """
 
+    def __init__(self, cache_dir, bucket_name):
```

Comment:
What I am asking ism should the constructor look like

```
def __init__ (self, cache_dir, bucket_name=None):
    if bucket_name is not None:
       self._bucket_name = bucket_name
```
so that we have the option of providing
```
class VBA2021CloudCache(S3CloudCache):
    _bucket_name = 'our-bucket-name`
```
for users who want to just use the `CloudCache` directly and not go through the full SDK API.

But maybe there are no such users?

Reply:
The idea being that those users would not want to have to know the name of the bucket.
Again: my mental model of data access may just be out of date.

Reply:
I don't much like having a class you can't instantiate without an inheriting class. I'd rather the low-level capabilities of `S3CloudCache` be usable.
I think your 2nd option is still possible, and, will probably be how I declare this:
```
class BehaviorProjectCloudApi(BehaviorProjectBase, S3CloudCache):
    def __init__(self, cache_dir=None, bucket_name="vba-2021-prototype"):
        ....
```
which in turn will be constructed from
```
class VisualBehaviorOphysProjectCache(Cache):
```
when `fetch_api` is a `BehaviorProjectCloudApi`
and will probably have a `from_cloud(cache_dir, bucket_name)` class or static method for instantiating,